### PR TITLE
Replace fake server with fakeDiscovery

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

This PR removes `createFakeServer` (cmd/support_archive/resources_test.go:350) to mock discovery response; instead, use FakeDiscovery client. Also extends k8sResourceCollector to pass DiscoveryClient and replace it inside readCustomResourceDefinitions instead of creating it inside method.

## How can this be tested?

unit tests
`make test/e2e/supportarchive`

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by aorcholski — CHANGES_REQUESTED** [review]

File: cmd/support_archive/resources_test.go [hunk 1]
```diff
@@ -132,15 +130,14 @@ func TestManifestCollector_Success(t *testing.T) {
...
 
-	apiResourceLists := getResourceLists()
-
-	server := createFakeServer(t, apiResourceLists[0], apiResourceLists[1], apiResourceLists[2])
+	client := fakeclientset.NewSimpleClientset()
 
-	defer server.Close()
-	rc := &rest.Config{Host: server.URL}
+	apiResourceLists := getResourceLists()
+	client.Resources = apiResourceLists[:3]
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply cmd/support_archive/resources_test.go:136]: good catch :)

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply cmd/support_archive/resources_test.go]: @aorcholski fixed in c11d0f76

File: cmd/support_archive/resources_test.go [hunk 2]
```diff
@@ -189,7 +185,10 @@ func TestManifestCollector_NoManifestsAvailable(t *testing.T) {
 
 	ctx := context.TODO()
```
Comment (aorcholski) [inline cmd/support_archive/resources_test.go:186]: IMHO context.Background() could be used directly in `newK8sObjectCollector(ctx...)` call. The variable is not reused .

**Review by aorcholski — CHANGES_REQUESTED** [review]

File: cmd/support_archive/resources_test.go (hunk 2 above)
Comment (andriisoldatenko) [reply cmd/support_archive/resources_test.go]: fixed in d485e00a

**Comment by aorcholski** [comment]
remove `csi.sqlite` file

**Pull request merged**